Sources/TripKit/helpers/TKUserProfileHelper.swift

```diff
@@ -132,5 +133,73 @@ public class TKUserProfileHelper: NSObject {
     }
   }
   
+  // MARK: - Mode by mode, Mode picker
+  
+  @objc public class func isSharedVehicleModeEnabled(identifier: ModeIdentifier) -> Bool {
+    return disabledSharedVehicleModes.contains(identifier)
+  }
+  
+  @objc public class func isSharedVehicleModeEnabled(mode: TKModeInfo) -> Bool {
+    var hasDisabled = false
+    disabledSharedVehicleModes.forEach { identifier in
+      let components = identifier.components(separatedBy: ":")
+      if components[0] == mode.identifier,
+         components[1] == mode.localImageName {
+        hasDisabled = true
+      }
+    }
+    return !hasDisabled
+  }
+  
+  /// update picked modes provided with all reference modes to determine disabled modes
+  @objc public class func update(pickedModes: Set<TKModeInfo>, allModes: Set<TKModeInfo>) {
+    let pickedIds = pickedModes.compactMap { TKUserProfileHelper.identifier(from: $0) }
+    let allIds = allModes.compactMap { TKUserProfileHelper.identifier(from: $0) }
+    update(pickedIds: pickedIds, allIds: allIds)
+  }
+  
+  @objc public class func update(pickedIds: [ModeIdentifier], allIds: [ModeIdentifier]) {
+    var modes = disabledSharedVehicleModes
+    
+    let toDisable = allIds.filter {
+      !pickedIds.contains($0)
+    }
+    
+    pickedIds.forEach { identifier in
+      if modes.contains(identifier) {
+        modes = modes.filter { identifier != $0 }
+      }
+    }
+    
+    toDisable.forEach {
+      if !modes.contains($0) {
+        modes.append($0)
+      }
+    }
+    
+    UserDefaults.shared.set(modes, forKey: DefaultsKey.disabled.rawValue)
+  }
+  
+  // combined both identifier and imageName to identify the actual mode, since identifier is the same for both bike and scooter modes
+  @objc public class func identifier(from mode: TKModeInfo) -> String? {
```

Comment:
Might be better to keep this one private, or is there a case where this need to be accessible?

```suggestion
  @objc private class func identifier(from mode: TKModeInfo) -> String? {
```

Reply:
ah yes, some of the methods here were previously used outside and decided to have it only in the class file